android-app/app/src/main/java/de/dhbw/ka/se/fibo/ui/settings/SettingsFragment.java

```diff
@@ -33,25 +33,34 @@
 public class SettingsFragment extends Fragment {
 
     private FragmentSettingsBinding binding;
+    private MaterialButton logoutButton;
     private MaterialButton deleteUserButton;
 
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         binding = FragmentSettingsBinding.inflate(inflater, container, false);
         View root = binding.getRoot();
 
         deleteUserButton = binding.deleteUser;
+        logoutButton = binding.logout;
 
         return root;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
-        initializeDeleteButton();
+        initializeButtons();
     }
 
-    private void initializeDeleteButton() {
+    private void initializeButtons() {
         deleteUserButton.setOnClickListener(e -> openDialog());
+        logoutButton.setOnClickListener(e -> logout());
+    }
+
+    private void logout() {
+        ApplicationState.getInstance(getContext()).clearAuthorization();
+        Intent i = new Intent(getActivity(), LoginActivity.class);
+        startActivity(i);
```

Comment:
```suggestion
          ActivityUtils.swapActivity(requireActivity(),LoginActivity.class,false);       
```

I added a #swapActivity method in ActivityUtils. It can be used if you want to change activities just like in that case. The benefit of the method is that you wont forget to enable or disable the allowBack flag. In the previous case the user could logout and press the back button. After that he would be logged out (no tokens) but could access the settings page again.

When I implemented the #swapActivity, I set the currentActivity paramter class to AppCompatActivity. Inorder to make this change work, you have to change the class from AppCompatActivity to just an Activity, because getActivity() does not provide a AppCompatActivity. It provides a FragmentActivity. The use of Activity allows both Fragment and AppCompatActivivities